### ISSUE TITLE
add the new block with 500 error

### ISSUE THREAD

**Issue by archonwang** [smartlime/redmine_unclosed_issues#1]
error log is 

```
Started GET "/my/page" for 192.168.60.238 at 2017-06-03 12:19:40 +0800
Processing by MyController#page as */*
  Current user: hui.wang (id=8)
  Rendered plugins/redmine_unclosed_issues/app/views/my/blocks/_unclosed_issues.html.erb (78.6ms)
  Rendered plugins/redmine_my_page_queries/app/views/my/page.html.erb within layouts/base (85.3ms)
Completed 500 Internal Server Error in 95ms (ActiveRecord: 3.3ms)

ActionView::Template::Error (Mysql2::Error: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ':conditions=>{:author_id=>8, :status_id=>3}}) FROM `issues` INNER JOIN `projects' at line 1: SELECT COUNT({:conditions=>{:author_id=>8, :status_id=>3}}) FROM `issues` INNER JOIN `projects` ON `projects`.`id` = `issues`.`project_id` WHERE ((projects.status <> 9 AND EXISTS (SELECT 1 AS one FROM enabled_modules em WHERE em.project_id = projects.id AND em.name='issue_tracking')) OR (projects.status = 1 AND EXISTS (SELECT 1 AS one FROM enabled_modules em WHERE em.project_id = projects.id AND em.name='issue_tracking')) OR (projects.status = 1 AND EXISTS (SELECT 1 AS one FROM enabled_modules em WHERE em.project_id = projects.id AND em.name='issue_tracking')))):
    1: <h3><%=l(:label_unclosed_issues)%> (<%= Issue.visible.count(:conditions => { :author_id => User.current.id, :status_id => 3 }) %>)</h3>
    2: 
    3: <% reported_issues = Issue.visible.
    4:         where(:author_id => User.current.id, :status_id => 3).
  plugins/redmine_unclosed_issues/app/views/my/blocks/_unclosed_issues.html.erb:1:in `_d662e294b065c0f8f221c0d5890f6dae'
  plugins/redmine_my_page_queries/app/helpers/my_page_queries_helper.rb:33:in `render_block'
  plugins/redmine_my_page_queries/app/views/my/page.html.erb:13:in `block in _ec5baa9bf2f89f2b16b33bc8fbcec2c7'
  plugins/redmine_my_page_queries/app/views/my/page.html.erb:10:in `each'
  plugins/redmine_my_page_queries/app/views/my/page.html.erb:10:in `_ec5baa9bf2f89f2b16b33bc8fbcec2c7'
  lib/redmine/sudo_mode.rb:63:in `sudo_mode'

```

**Comment by smartlime** [comment]
It looks like `.where()` doesn't works for you, strange but looks like this. What Redmine and Rails version do you use?

**Comment by archonwang** [comment]
@smartlime 

> Environment:
>   Redmine version                3.3.3.stable.16676
>   Ruby version                   2.2.4-p230 (2015-12-16) [x86_64-linux]
>   Rails version                  4.2.7.1
>   Environment                    production
>   Database adapter               Mysql2

**Comment by smartlime** [comment]
As this plugin was created long ago it is wasn't tested with Rails 4.2. I fix counter at latest master (a0207da) but I'm not sure the rest of code will work with modern Rails.

**Comment by archonwang** [comment]
thanks. it worked now.
by the way, could it config the issue status which to be closed? Thanks.
